src/components/panel/panel.spec.js

```diff
@@ -643,6 +643,227 @@ describe('$mdPanel', function() {
       expect(scrollMaskEl).not.toExist();
       expect($mdUtil.disableScrollAround).toHaveBeenCalled();
     });
+
+    describe('animation hooks: ', function() {
+      it('should call onDomAdded if provided when adding the panel to the DOM',
+          function() {
+            var onDomAdded = jasmine.createSpy();
+            var config = angular.extend(
+                {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+            attachPanel(config);
+
+            expect(onDomAdded).toHaveBeenCalled();
+            expect(PANEL_EL).toExist();
+          });
+
+      it('onDomAdded should return a promise and continue resolving the ' +
+          'rest of promise chain', function() {
+            var onDomAdded = jasmine.createSpy();
+            var config = angular.extend(
+                {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+            openPanel(config);
+
+            expect(onDomAdded).toHaveBeenCalled();
+            expect(PANEL_EL).toExist();
+            expect(PANEL_WRAPPER_CLASS).not.toHaveClass(HIDDEN_CLASS);
+          });
+
+      it('should onDomAdded reject, the panel should not be shown', function() {
+
+        var onDomAdded = function() {
+          return panelRef._$q.reject();
+        };
+
+        var config = angular.extend(
+            {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+        panelRef = $mdPanel.create(config);
+
+        var openRejected = false;
+
+        panelRef.show = function() {
+          return panelRef._$q.reject();
```

Comment:
You aren't testing what the description says you are. You want to test what happens when `onDomAdded` is rejected. As written, you are testing what happens when `show` rejects. This also applies to the other tests in this section.

For the returning a promise tests, onDomAdded needs to be a promise that resolves (`$q.when(true)`), not a spy. For the rejecting a promise tests, onDomAdded needs to be a promise that rejects (`$q.reject()`).